Source/WebKit/UIProcess/API/C/mac/WKPagePrivateMac.mm

```diff
@@ -55,61 +55,67 @@ - (id)initWithPage:(RefPtr<WebKit::WebPageProxy>&&)page
         return nil;
 
     _page = WTFMove(page);
-    _observer = WebKit::PageLoadStateObserver::create(self, @"URL");
-    _page->pageLoadState().addObserver(*_observer);
+    Ref observer = WebKit::PageLoadStateObserver::create(self, @"URL");
+    _observer = observer.get();
+    self._protectedPage->protectedPageLoadState()->addObserver(observer.get());
 
     return self;
 }
 
+- (Ref<WebKit::WebPageProxy>)_protectedPage
```

Comment:
@darinadler mentioned in earlier reviews that he prefers us using a static C++ function rather than a ObjC method like this (Please see WKPreferences.mm as example).

Reply:
The reasons I suggested functions were:
1) They can be private to the source file when they are used only in one file, typically the implementation file of the class in question.
2) They have less runtime overhead than a method since there is no method dispatch and are eligible to be inlined by the compiler.
If those reasons don’t apply, we could still use an Objective-C property or method.